src/algebra/gcd_monoid/basic.lean

```diff
@@ -540,6 +540,21 @@ begin
   exact associated_of_dvd_dvd (gcd_monoid.dvd_gcd hda hdb) h,
 end
 
+lemma extract_gcd {α : Type*} [cancel_comm_monoid_with_zero α] [gcd_monoid α] (x y : α) :
```

Comment:
I think we can extract a convenient sublemma here: if `x = x' * (gcd x y)` and `y = y' * (gcd x y)`, then `is_unit (gcd x' y')`.